libs/langchain/langchain/vectorstores/cratedb/model.py

```diff
@@ -62,6 +62,19 @@ def get_by_name(
                         raise
                 return None
 
+            @classmethod
+            def get_by_names(
+                cls, session: Session, names: List[str]
+            ) -> Optional["List[CollectionStore]"]:
+                try:
+                    return (
+                        session.query(cls).filter(cls.name.in_(names)).all()  # type: ignore[attr-defined]  # noqa: E501
+                    )
+                except sqlalchemy.exc.ProgrammingError as ex:
+                    if "RelationUnknown" not in str(ex):
```

Comment:
if a table doesn't exist it doesn't fail?

Reply:
Thank you for spotting this. I've just added 0f154e687c161fb to get this more straight by moving the exception handling to the very spot where it is needed, accompanied by extended inline comments.